### ISSUE TITLE
Image field error on image upload

### ISSUE THREAD

**Issue by edueter** [tinacms/tinacms#1084]
### Intro
In the docs about the image field, there's a proper way to manage images within Gatsby. I've even changed my setup so the Markdown files would appear as index.md in folders name with slugs. i.e.: /blog/hello-world/index.md, matching the code from this page.

### What is expected
To upload images into posts that are already there.

### What is happening
I'm getting this error as I upload any image. I'm really not sure what's wrong.
![Capturar](https://user-images.githubusercontent.com/39069806/80811965-6216a900-8b9d-11ea-92cb-44a7cbd0f96e.PNG)

### The code
You can follow [this link](https://github.com/edueter/gatsby-theme-cripto/blob/Protoype-TinaCMS/src/templates/blogPost.js) to blogPost.js template for the full code, but here it is the head and component for TinaCMS.

```
/*
** src/template/blogPost.js
*/

import React from "react"
import { graphql, Link } from "gatsby"

import { remarkForm, DeleteAction } from 'gatsby-tinacms-remark'
import get from 'lodash.get'

import Img from "gatsby-image"
import AniLink from "gatsby-plugin-transition-link/AniLink"

import BlogLayout from '../components/layout/BlogLayout'
import { SEO } from 'gatsby-theme-seo'

import kebabCase from "lodash/kebabCase"

import TextScramble from "../components/TextScramble"

( ... template code ... )

const path = require('path').posix

 // 1. Define the form config
 const BlogPostForm = {
  label: 'Postagem de blog',
  fields: [
    {
      label: 'Título',
      name: 'rawFrontmatter.title',
      description: 'Insira o título do post aqui',
      component: 'text',
    },
    {
      label: 'Data de publicação',
      name: 'rawFrontmatter.date',
      description: 'Insira a data de publicação',
      component: 'date',
      dateFormat: 'DD MMMM YYYY',
      timeFormat: 'false',
    },
    {
       name: "rawFrontmatter.draft",
       component: "toggle",
       label: "Rascunho"
    },
    //{
    //  name: 'rawFrontmatter.autor',
    //  component: 'select',
    //  label: 'Autor',
    //  description: 'Selecione um autor para essa publicação',
    //  options: ['Instituto Beta', 'Adriana Melo', 'Mateus Linhares', 'Renato Vasconcelos', 'Vanessa Lage'],
    //},
    //{
    //  label: 'Autor',
    //  name: 'rawFrontmatter.autor',
    //  component: 'author',
    //},
    {
      label: 'Imagem do post',
      name: 'rawFrontmatter.featuredImage',
      description: 'Dê preferências para imagens de alta resolução. Um lugar bom para se procurar tais imagens, é o https://unsplash.com/.',
      component: 'image',
      previewSrc: (formValues, { input }) => {
        const path = input.name.replace('rawFrontmatter', 'frontmatter')
        const gatsbyImageNode = get(formValues, path)
        if (!gatsbyImageNode) return ''
        //specific to gatsby-image
        return gatsbyImageNode.childImageSharp.fluid.src
      },

      // upload images to same directory as content file
      uploadDir: blogPost => path.dirname(blogPost.fileRelativePath), // ** This is the line my browser is point out.

      // image file is sibling of content file
      parse: filename => `./${filename}`,
    }, // Imagem do post
    {
      name: 'rawMarkdownBody',
      component: 'markdown',
      label: 'Conteúdo',
      description: 'Edite o conteúdo do post aqui',
    },
  ],
  actions: [DeleteAction],
}

export default remarkForm(Template, BlogPostForm)

( ... queries ... )
```

Any help or tip would be much appreciated. I've gone browsing about this but couldn't find anything that would help me out with this.

**Comment by ncphillips** [comment]
I believe the issue is this:
```
const path = require('path').posix
```
`path` is a Node module so it's not available in the browser. 

There may be a browser-compatible package you could use instead. Orr you could make your own `dir` function:

```
function dir(path: string) {
    let fileParts = path.split('/')
    let dirParts = fileParts.slice(0, parts.length - 1)
    return dirParts.join('/')
}

// ...

uploadDir: blogPost => dir(blogPost.fileRelativePath)
```

**Comment by edueter** [comment]
Oh, I thought it would be a mandatory const with that. Sorry.

I've tried your suggestion and it returned me an error. Failed to compile because "Type annotation can only be used in TypeScript" files, so I've added support to TypeScript files (just for sake of curiosity). That way, Gatsby develop runs fine but when I try to upload an image, I get this couple of errors. Gonna take a look into it after lunch.
![Capturar](https://user-images.githubusercontent.com/39069806/80817281-85465600-8ba7-11ea-92cd-1d7681f223d1.PNG)


Thanks very much for your comment, @ncphillips
